internal/data/spicedb.go

```diff
@@ -149,6 +149,32 @@ func (s *SpiceDbRepository) DeleteRelationships(ctx context.Context, filter *api
 	return nil
 }
 
+func (s *SpiceDbRepository) CheckPermission(ctx context.Context, check *apiV1.CheckPermissionRequest) (*apiV1.CheckPermissionResponse, error) {
+	subject := &v1.SubjectReference{Object: &v1.ObjectReference{
+		ObjectType: check.Subject,
+		ObjectId:   check.Subjectid,
+	}}
+
+	object := &v1.ObjectReference{
+		ObjectType: check.Resourcetype,
+		ObjectId:   check.Resourceid,
+	}
+	checkResponse, err := s.client.CheckPermission(ctx, &v1.CheckPermissionRequest{
+		Resource:   object,
+		Permission: check.Operation,
+		Subject:    subject,
+	})
+	if err != nil {
```

Comment:
@akoserwal should this err be somehow logged ? (to see reason of err)

Reply:
Thanks, @lpichler I will update